Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift

```diff
@@ -73,9 +74,8 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
     }
 
     deinit {
-        // TODO: we could wrap the failure here with diagnostics if it wasn't optional throughout
         guard case .disconnected = (try? self.withStateLock { self.state }) else {
-            return assertionFailure("db should be closed")
+            return self.observabilityScope.emit(error: "db should be closed")
```

Comment:
@yim-lee are you okay with this change or prefer we continue to assert?

Reply:
(could probably use a better error message)

Reply:
I'm ok with logging rather than failure.